library/ZendService/Twitter/Response.php

```diff
@@ -51,21 +57,23 @@ class Response
      * @param  HttpResponse $httpResponse
      * @throws Exception\DomainException if unable to decode JSON response
      */
-    public function __construct(HttpResponse $httpResponse)
+    public function __construct(HttpResponse $httpResponse = null)
     {
         $this->httpResponse = $httpResponse;
-        $this->rawBody      = $httpResponse->getBody();
-        try {
-            $jsonBody = Json::decode($this->rawBody, Json::TYPE_OBJECT);
-            $this->jsonBody = $jsonBody;
-        } catch (JsonException $e) {
-            throw new Exception\DomainException(sprintf(
-                'Unable to decode response from twitter: %s',
-                $e->getMessage()
-            ), 0, $e);
+
+        if (! is_null($httpResponse) and ! empty($httpResponse->getBody())) {
+            $this->populate($httpResponse);
         }
     }
 
+    protected function setRateLimit($headers = [])
```

Comment:
All other delegate methods in the class are protected, so this is actually consistent.